### PR TITLE
[FEATURE] Improve Error Handling and Validation for status Attribute in Task Class

### LINKED ISSUE #18: [FEATURE] Improve Error Handling and Validation for `status` Attribute in `Task` Class
### Description

**Severity/Priority:** 
- **Low** 🟢

**Status:** 
- **Accepted** 👍

The current error handling and validation of the `status` attribute in `Task` objects should be simplified so that the caller handles an exception raised while setting a `status`. This allows for more predictable behavior and data integrity by not forcing a potentially inaccurate default or fallback value.

---

## Motivation

A `Task` object maintains a list of valid `status` attributes. Current behavior performs a convoluted check that sets a default value of "Pending" if an invalid status is passed. This can cause confusion over the validity of a status (if it was *set* pending or *error/defaulted* to pending).

Improved error handling will allow the caller to choose how to handle an invalid status and not force a potentially confusing default status onto the `Task` object.
 
---

## Proposed Solution

The validation logic can be simplified to only the setter method, merging in the `_validate_status()` checks. This simplifies the process to a check against the list of valid statuses. If there is not a match, then an exception is raised (`InvalidStatusError`), allowing the caller to decide how to handle the error.

If no status is provided during initialization, the default value of "Pending" will be used to ensure backward compatibility.

---

## Alternatives Considered

- Keep default "Pending" behavior

---

## Impact

The improved validation and error handling will provide predictable behavior and better data integrity (the status was explicitly set, not an error fallback).

Backwards compatibility is preserved by providing a default `status` of "Pending" if a `status` is not explicitly provided during construction.

---

### PR DESCRIPTION
**Pull request by cmorman89**
## 🔄 Summary of Changes

**Type of Change:** New Feature 🌱

The `status` validation in the `Task` class has been simplified. The `_validate_status()` method has been removed, and the logic moved into the `@status.setter` method. This method checks the provided `new_status` against a list of `valid_statuses` to ensure it is set to a valid status. If the status is invalid, it raises `InvalidStatusError`.

The default status remains as "Pending" if no status is provided during initialization to maintain consistent behavior.`.

---

## ✅ Checklist
- [x] My code follows the project's style guidelines.
- [x] I have made corresponding changes to the documentation.
- [x] I have added tests that prove my changes are effective.
- [x] All new and existing tests pass.

---

### Motivation

A Task object maintains a list of valid status attributes. Current behavior performs a convoluted check that sets a default value of "Pending" if an invalid status is passed. This can cause confusion over the validity of a status (if it was set pending or error/defaulted to pending).

Improved error handling will allow the caller to choose how to handle an invalid status and not force a potentially confusing default status onto the Task object.

### Proposed Solution

- Merge `_validate_status()` method logic into `@status.setter` method:
  - The combined method performs the same check against the internal `valid_status` list as before.
  - Pass validation: set the `Task` status to the value passed.
  - Fail validation: raise `InvalidStatusError`
- Backward compatability:
  - Retain default `status` value "Pending" if not set during `init`.
- Unit tests:
  - Test setting valid statuses.
  - Test raising an exception when setting invalid statuses and edge cases.
 
---

## 🧪 Testing

- **Test Cases:**
  1. **Case 1:** `test_task_valid_status_setter`
    - Test parameterized, valid statuses in regular, upper, and lower cases.
      - Param Example: `"Pending"`, `"In Progress"`, `"Complete"`
      - Expected: `status == valid_status`

  2. **Case 2:** `test_task_invalid_status_setter`
    - Test parameterized, invalid statuses
      - Param Example: `"Finished"`, `"Completed"`, `""`, `"\n"`
      - Expected: `raise InvalidStatusError`

  3. **Case 3:** `test_task_construction`
   - Test initializing a `Task` with no `status` argument.
      - Expected: `status == "Pending"`
    - Test initializing a `Task` with a valid `status` argument.
      - Expected: `status == valid_status`
  

### Test Environment
- **OS:**  Ubuntu 22.04 (via WSL2 on Windows 11)
- **Python Version:**  Python 3.13.0

---

## 📎 Related Issues

Closes #18 

---

**Additional Notes**

> [!NOTE]
> Valid statuses are:
> - `"Pending"`
> - `"In Progress"`
> - `"Complete"`

### CODE REVIEW

File: app/task.py [hunk 1]
```diff
@@ -31,23 +33,38 @@ def __init__(self, task_id: int):
         super().__init__(f"Task name cannot be blank for task ID #{task_id}.")
         self.task_id = task_id
 
+
+class InvalidStatusError(TaskError):
+    """Raise when attempting to set a status that is not in the list of valid statuses.
+
+    Attributes:
+        task_id: The ID of the task that raised the exception.
```
Comment (cmorman89) [inline app/task.py:41]: Typing: `int`

File: app/task.py [hunk 2]
```diff
@@ -31,23 +33,38 @@ def __init__(self, task_id: int):
...
+        attempted_status: The invalid status attempted by the user.
```
Comment (cmorman89) [inline app/task.py:42]: Typing: `str`

File: app/task.py [hunk 3]
```diff
@@ -31,23 +33,38 @@ def __init__(self, task_id: int):
...
+        valid_statuses: The list of valid statuses that may be set.
```
Comment (cmorman89) [inline app/task.py:43]: Typing: `List[str]`

File: app/task.py [hunk 4]
```diff
@@ -31,23 +33,38 @@ def __init__(self, task_id: int):
...
+
+
 class Task:
     """Responsible for holding task-related information such as an ID, title, due date, etc.
 
     Attributes:
         task_id (int): The unique ID of the task, automatically generated using
             `TaskID.generate()` during instantiation. Immutable once set.
-        title (str): A short title or description of the task.
+        title (str): A short title or description of the task. Must not be blank or `None.
```
Comment (cmorman89) [inline app/task.py:61]: Missing backtick ``` ` ``` after `None`
